dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/engine/command/impl/CuratePublishedDatasetVersionCommand.java

```diff
@@ -125,8 +125,6 @@ public Dataset execute(CommandContext ctxt)  {
                     metadataUpdated = true;
                 }
 
-            } else {
-                throw new IllegalCommandException("Cannot change files in the dataset", this);
```

Comment:
IMO something not pretty could happen in some scenarios.
Please note that we iterating over dataset.getFiles() - this will return all files for dataset - regardless of it's version.
Lets assume that we have a dataset with 3 versions:
V1: file1.txt, file2.txt
V2: file1.txt
DRAFT: file1.txt

In such cases variables:
`FileMetadata draftFmd = dataFile.getLatestFileMetadata();` == V1: file2.txt
`FileMetadata publishedFmd` == null (`fmd.getDatasetVersion().equals(updateVersion)` will always be false in loop)

We removed else statement so code below your change will be executed:
```
FileMetadata mergedFmd = ctxt.em().merge(draftFmd);
ctxt.em().remove(mergedFmd);
```
So we removed filemetadata from V1 - which should be untouched